src/main/java/com/sonyericsson/jenkins/plugins/bfa/db/DynamoDBKnowledgeBase.java

```diff
@@ -0,0 +1,534 @@
+        if (dbMapper != null) {
+            return dbMapper;
+        }
+        dbMapper = new DynamoDBMapper(getDynamoDb());
+        createTable(dbMapper.generateCreateTableRequest(FailureCause.class));
+
+        return dbMapper;
+    }
+
+    /**
+     * Creates a DynamoDB table.
+     * @param request {@link CreateTableRequest}
+     */
+    private void createTable(CreateTableRequest request) {
+        try {
+            String tableName = request.getTableName();
+            AmazonDynamoDB db = getDynamoDb();
+            request.setProvisionedThroughput(new ProvisionedThroughput()
+                    .withReadCapacityUnits(1L).withWriteCapacityUnits(1L));
+            TableUtils.createTableIfNotExists(db, request);
+            TableUtils.waitUntilActive(db, tableName);
+
+        } catch (Exception e) {
+            throw new AmazonClientException(e);
+        }
+    }
+
+    /**
+     * Use Jenkins to get and instance of {@link DynamoDBKnowledgeBaseDescriptor}.
+     * @return Descriptor
+     */
+    @Override
+    public Descriptor<KnowledgeBase> getDescriptor() {
+        return Jenkins.getInstance().getDescriptorByType(DynamoDBKnowledgeBaseDescriptor.class);
+    }
+
+    /**
+     * Descriptor for {@link DynamoDBKnowledgeBase}.
+     */
+    @Extension
+    public static class DynamoDBKnowledgeBaseDescriptor extends KnowledgeBaseDescriptor {
+
+        @Override
+        public String getDisplayName() {
+            return Messages.DynamoDBKnowledgeBase_DisplayName();
+        }
+
+        /**
+         * Convenience method for jelly.
+         * @return the default region.
+         */
+        public String getDefaultRegion() {
+            return DYNAMODB_DEFAULT_REGION;
+        }
+
+        /**
+         * Convenience method for jelly.
+         * @return the default region.
+         */
+        public String getDefaultCredentialsPath() {
+            return DYNAMODB_DEFAULT_CREDENTIALS_PATH;
+        }
+
+        /**
+         * Convenience method for jelly.
+         * @return the default region.
+         */
+        public String getDefaultCredentialProfile() {
+            return DYNAMODB_DEFAULT_CREDENTIAL_PROFILE;
+        }
+
+        /**
+         * Get a list of valid AWS regions for Jelly.
+         * @return ListBoxModel containing AWS regions
+         */
+        public ListBoxModel doFillRegionItems() {
+            ListBoxModel items = new ListBoxModel();
+            for (Region r:RegionUtils.getRegions()) {
+                String regionName = r.getName();
+                items.add(regionName, regionName);
+            }
+            return items;
+        }
+
+        /**
+         * Checks that the credential file exists.
+         *
+         * @param value the pattern to check.
+         * @return {@link hudson.util.FormValidation#ok()} if everything is well.
+         */
+        public FormValidation doCheckCredentialsPath(@QueryParameter("value") final String value) {
```

Comment:
first off this should use proper credentials via the credentials store instead ( the old mongodb code was done before we had any real knowledge of how Jenkins credentials worked :) )

Second this is a potential security risk as it can be used by anyone with overall/read permission to query the existence of a file on the Jenkins master file system. So needs `@RequirePOST` and a `Jenkins.checkPermission(Jenkins.ADMINISTER)`. The jelly also needs to be updated so the check is done via POST instead of a GET.